sourcemap/mapper.go

```diff
@@ -63,5 +64,11 @@ func (m *SmapMapper) Apply(id Id, lineno, colno int) (*Mapping, error) {
 }
 
 func (m *SmapMapper) NewSourcemapAdded(id Id) {
+	_, err := m.Accessor.Fetch(id)
+	sourcemapErr, ok := err.(Error)
+	if err == nil || (ok && sourcemapErr.Kind == MapError) {
```

Comment:
I have looked at the `Fetch` method again, and here a check for `err == nil` is sufficient. 
The only two possibilities that no error is returned is when the sourcemap is found in the cache or found in ES.

However, if a nil sourcemap is cached or nothing is found in ES but no error was thrown, then a `MapError` is returned from `Fetch`, but in this case there should not be logged a warning here. 

To summarize: Your first attempt was perfectly fine, sorry for my misleading comment.

Reply:
Damn, its hard to think about this. `Fetch` could also return `KeyError`, but it doesn't matter because in that case shouldn't be logged neither.